WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -0,0 +1,178 @@
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable(WordPress.SITE, mSite);
+        outState.putString(KEY_PLUGIN_NAME, mPlugin.getName());
+    }
+
+    private void setupViews() {
+        mSwitchActive = findViewById(R.id.plugin_state_active);
+        mSwitchAutoupdates = findViewById(R.id.plugin_state_autoupdates);
+
+        mSwitchActive.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                if (compoundButton.isPressed()) {
+                    mPlugin.setIsActive(b);
+                    dispatchUpdateAction();
+                }
+            }
+        });
+
+        mSwitchAutoupdates.setOnCheckedChangeListener(new OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                if (compoundButton.isPressed()) {
+                    mPlugin.setIsAutoUpdateEnabled(b);
+                    dispatchUpdateAction();
+                }
+            }
+        });
+
+        findViewById(R.id.plugin_wp_org_page).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ActivityLauncher.openUrlExternal(PluginDetailActivity.this, getWpOrgPluginUrl());
+            }
+        });
+
+        findViewById(R.id.plugin_home_page).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ActivityLauncher.openUrlExternal(PluginDetailActivity.this, mPlugin.getPluginUrl());
+            }
+        });
+
+        refreshStates();
+    }
+
+    private void refreshStates() {
+        mSwitchActive.setChecked(mPlugin.isActive());
+        mSwitchAutoupdates.setChecked(mPlugin.isAutoUpdateEnabled());
+    }
+
+    private void dispatchUpdateAction() {
+        mDispatcher.dispatch(PluginActionBuilder.newUpdateSitePluginAction(
+                new UpdateSitePluginPayload(mSite, mPlugin)));
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSitePluginUpdated(OnSitePluginUpdated event) {
+        if (event.isError()) {
+            if (event.error.type == UpdateSitePluginErrorType.ACTIVATION_ERROR
+                    || event.error.type == UpdateSitePluginErrorType.DEACTIVATION_ERROR) {
+                // these errors are thrown when the plugin is already active and we try to activate it and vice versa.
+                return;
+            }
+            ToastUtils.showToast(this, "An error occurred while fetching the plugins: "
+                    + event.error.message);
+            return;
+        }
+        if (event.plugin != null) {
+            mPlugin = event.plugin;
+            refreshStates();
+        }
+    }
+
+    // Helpers
```

Comment:
I think the other `private` methods can be considered `Helpers`, nbd though ;)

Reply:
I would consider the other ones just UI methods and not really helpers. If I move those here, it breaks the flow of the code. We could add another title to those ones or just remove this one if you'd like. What do you think?

Reply:
Nah, no need to move things around here it was a really nitpicky comment anyway.

Reply:
Thanks for bringing it up anyway! I am always open to ideas to improve the style 👍